devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -445,9 +446,16 @@ private Xpp3Dom getPluginConfig(Plugin plugin, String goal) {
 
         final Xpp3Dom configuration = configuration();
         if (mergedConfig != null) {
+            Set<String> supportedParams = null;
             // Filter out `test*` configurations
             for (Xpp3Dom child : mergedConfig.getChildren()) {
-                if (!child.getName().startsWith("test")) {
+                if (child.getName().startsWith("test")) {
+                    continue;
+                }
+                if (supportedParams == null) {
+                    supportedParams = getMojoDescriptor(plugin, goal).getParameterMap().keySet();
```

Comment:
Is there any value in lazy loading the `supportedParams`? Looks like unneeded complexity to me but maybe I miss something?

Reply:
Not much value. E.g. in case the configuration elements are present but empty.